go/libraries/doltcore/remotestorage/chunk_store.go

```diff
@@ -884,7 +863,7 @@ func (dcs *DoltChunkStore) Close() error {
 
 // getting this working using the simplest approach first
 func (dcs *DoltChunkStore) uploadChunks(ctx context.Context) (map[hash.Hash]int, error) {
-	hashToChunk := dcs.cache.GetAndClearChunksToFlush()
+	hashToChunk := dcs.wb.GetAllAndClear()
```

Comment:
Actually, clearing this buffer here races with concurrent access on read-your-own-writes and fails to be robust with regards to the presence of the written data if we encounter an error on the upload. AFAICT, both of these problems also exist in the current implementation.